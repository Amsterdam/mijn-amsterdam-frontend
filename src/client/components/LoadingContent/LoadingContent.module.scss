// Concept copied from https://github.com/yarism/react-loading-placeholder/blob/master/src/LoadingPlaceholder.css
@keyframes placeHolderShimmer {
  0% {
    background-position: -1168px 0;
  }
  100% {
    background-position: 1168px 0;
  }
}

.LoadingContent {
  display: block;
  > span {
    display: block;
    height: 1rem;
    margin-bottom: 1rem;
    animation-duration: 3s;
    animation-fill-mode: forwards;
    animation-iteration-count: infinite;
    animation-name: placeHolderShimmer;
    animation-timing-function: linear;
    background: $color-neutral-grey4;
    background: linear-gradient(
      to right,
      $color-neutral-grey3 8%,
      $color-neutral-grey4 18%,
      $color-neutral-grey3 33%
    );
    background-size: 1200px 104px;
    border-radius: 2px;
    opacity: 0.2;
  }
}
