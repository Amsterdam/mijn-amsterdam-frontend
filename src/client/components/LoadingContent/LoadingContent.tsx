import { ComponentChildren } from '../../../universal/types';
import React from 'react';
import classnames from 'classnames';
import styles from './LoadingContent.module.scss';

type width = string;
type height = string;
type marginBottom = string;

export type BarConfig = Array<[width, height, marginBottom]>;

export interface ComponentProps {
  children?: ComponentChildren;
  barConfig?: BarConfig;
  className?: string;
}

const defaultBarConfig: BarConfig = [
  ['70%', '2rem', '.7rem'],
  ['90%', '2rem', '.7rem'],
  ['90%', '2rem', '.7rem'],
  ['40%', '2rem', '0'],
];

export default function LoadingContent({
  barConfig = defaultBarConfig,
  className,
}: ComponentProps) {
  return (
    <span className={classnames(styles.LoadingContent, className)}>
      {barConfig.map(([width, height, marginBottom], index) => {
        return (
          <span
            key={index}
            style={{
              width,
              height,
              marginBottom,
            }}
          />
        );
      })}
    </span>
  );
}
