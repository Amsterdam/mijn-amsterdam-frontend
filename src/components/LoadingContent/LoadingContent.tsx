import React from 'react';
import styles from './LoadingContent.module.scss';
import { ComponentChildren } from 'App.types';
import classnames from 'classnames';

export type BarConfig = Array<string[]>;

export interface ComponentProps {
  children?: ComponentChildren;
  barConfig?: BarConfig;
  className?: any;
}

const defaultBarConfig = [
  ['70%', '2rem', '.7rem'],
  ['90%', '2rem', '.7rem'],
  ['90%', '2rem', '.7rem'],
  ['40%', '2rem', '0'],
];

export default function LoadingContent({
  children,
  barConfig = defaultBarConfig,
  className,
}: ComponentProps) {
  return (
    <div className={classnames(styles.LoadingContent, className)}>
      {barConfig.map(([width, height, marginBottom], index) => {
        return (
          <div
            key={index}
            style={{
              width,
              height,
              marginBottom,
            }}
          />
        );
      })}
    </div>
  );
}
