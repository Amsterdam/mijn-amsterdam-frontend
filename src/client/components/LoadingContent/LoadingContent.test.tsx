import { render } from '@testing-library/react';

import LoadingContent from './LoadingContent.tsx';

describe('LoadingContent component', () => {
  it('Renders without crashing', () => {
    render(<LoadingContent />);
  });
});
