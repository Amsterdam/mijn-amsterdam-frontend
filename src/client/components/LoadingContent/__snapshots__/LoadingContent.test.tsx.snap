// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContent component Show the correct custom barConfig 1`] = `"<span class=\\"LoadingContent\\"><span style=\\"width:300px;height:20px;margin-bottom:10px\\"></span><span style=\\"width:200px;height:10px;margin-bottom:5px\\"></span></span>"`;

exports[`LoadingContent component Show the correct default barConfig 1`] = `"<span class=\\"LoadingContent\\"><span style=\\"width:70%;height:2rem;margin-bottom:.7rem\\"></span><span style=\\"width:90%;height:2rem;margin-bottom:.7rem\\"></span><span style=\\"width:90%;height:2rem;margin-bottom:.7rem\\"></span><span style=\\"width:40%;height:2rem;margin-bottom:0\\"></span></span>"`;
